waku/node/peer_manager/peer_manager.nim

```diff
@@ -619,30 +606,59 @@ proc getNumStreams*(pm: PeerManager, protocol: string): (int, int) =
   return (numStreamsIn, numStreamsOut)
 
 proc pruneInRelayConns(pm: PeerManager, amount: int) {.async.} =
-  let (inRelayPeers, outRelayPeers) = pm.connectedPeers(WakuRelayCodec)
+  let (inRelayPeers, _) = pm.connectedPeers(WakuRelayCodec)
   let connsToPrune = min(amount, inRelayPeers.len)
 
   for p in inRelayPeers[0..<connsToPrune]:
     asyncSpawn(pm.switch.disconnect(p))
 
 proc connectToRelayPeers*(pm: PeerManager) {.async.} =
-  let (inRelayPeers, outRelayPeers) = pm.connectedPeers(WakuRelayCodec)
-  let maxConnections = pm.switch.connManager.inSema.size
-  let totalRelayPeers = inRelayPeers.len + outRelayPeers.len
-  let inPeersTarget = maxConnections - pm.outRelayPeersTarget
+  var peersToConnect: seq[RemotePeerInfo]
+  var peersToDisconnect: int
+
+  for shard in pm.wakuMetadata.shards.items:
```

Comment:
not sure that we can use `wakuMetadata.shards` right now. afaik `shards` is not populated nor updated yet?

Reply:
The PRs are splitted but I intent to merge the shard subscription one first.